### DISCUSSION TITLE
Comparing between antiSMASH 5 and antiSMASH 7 data

### DISCUSSION THREAD

**Discussion by MWMullowney** [antismash/antismash#772] [General]
I am working on a project to compare about 100 genomes to the antiSMASH dataset included in the [BiG-SLiCE paper](https://academic.oup.com/gigascience/article/10/1/giaa154/6092777). It's a massive dataset that was generated using antiSMASH version 5.1.1 on genomes from NCBI. I have looked up the changelog for antiSMASH between 5.1.1 and 7.1, which is quite informative. Still I'm not clear on how much my comparison results (using BiG-SLICE and potentially other tools) will be biased by the updates in the software (newly detected BGCs, updates to detection rules, etc.) rather than actual differences in the datasets. I don't have the resources or time to rerun the entire NCBI dataset in antiSMASH 7 so that it matches my data, and I don't think regressing to antiSMASH 5 to reanalyze my data to match the BiG-SLiCE dataset is a good idea. Does anyone have a sense for what portion of newly detected BGCs in my dataset might be solely from the updates to antiSMASH rather than from the fact that my genomes might have unique BGCs? Thanks for your thoughts!

**Comment by zdk123** [comment]
The latest [antiSMASH-DB](https://antismash-db.secondarymetabolites.org/) has v7.1.0 annotations for RefSeq (the release is a year old now - but still).

**Reply by MWMullowney** [reply]
Oh, right, I should have thought of this! Do you know if it's possible to download all bacterial BGCs from the antiSMASH database?

**Reply by zdk123** [reply]
the full archive is [here](https://dl.secondarymetabolites.org/database/4.0/) but if you just need the 100 subset you could grab them off the server directly for a given refseq assembly accession at `https://antismash-db.secondarymetabolites.org/output/{ACC}/{ACC}.zip`

**Reply by MWMullowney** [reply]
Thanks a million, Zachary. This is so helpful.

**Reply by MWMullowney** [reply]
Hi Zach, do you know how to open the database files? I can't seem to figure out how to extract the BGC info. I was expecting it might be a collection of gbk files, but it is a SQL and two JSON files. Is there a specific tool that enables extracting these BGCs to use as input in BiG-SLiCE?

**Reply by zdk123** [reply]
The `asdb-beta{1,2}-jsons.tar` files contain all the individual antismash json files. I think you'd have to play around with antismash to extract the BGC regions as genbanks.

**Reply by zreitz** [reply]
For antiSMASH, you'll want the `--reuse-results` flag, which will regenerate the results folder. I would think you'd also want the `--minimal` flag to save time, since BiG-SLICE doesn't need much info.

You might find my wrapper [multiSMASH](https://github.com/zreitz/multismash/) useful here. There's a [reuse_results](https://github.com/zreitz/multismash/blob/master/example/config-example.yaml#L46) flag that I added last time I worked with aSDB JSONs. multiSMASH is a WIP, so let me know in that repo if you go this route and need a hand.

**Reply by MWMullowney** [reply]
Thank you both! Our team is going to try these ideas. 🤞🏻

**Comment by MWMullowney** [comment]
Hi Zach (both of you), upon downloading the antiSMASH database, it looks
like the two tar files (“beta1” and “beta2”) contain redundant json files
for a bunch of the genomes. It looks like one of each of the redundant
pairs is just an updated output. Do you know anything by about this? Maybe
they left old version antiSMASH outputs in with the newer version 7 ones?
My plan is to just run the “reuse-results” tag on the most up-to-date
output and ignore the older one for those that have two. Do you know of any
reason I should hang on to the older file?

**Comment by MWMullowney** [comment]
...disregard! I found the answer in the antismash db paper: "After creating
the updated datasets, antiSMASH 7 was re-run on the first round's results
using the options ‘cb-general reuse’, also updating CompaRiPPson results
with the new dataset." (
https://academic.oup.com/nar/article/52/D1/D586/7334085)
